apps/prairielearn/src/ee/lib/ai-grading.ts

```diff
@@ -212,22 +193,41 @@ async function ensureSubmissionEmbedding({
   );
   const $ = cheerio.load(render_submission_results.data.submissionHtmls[0], null, false);
   $('script').remove();
-  const student_answer = $.html();
-  const embedding = await createEmbedding(openai, student_answer, `course_${course.id}`);
-  // insert new embedding into the table and return the new embedding
+  const submission_text = $.html();
+  const embedding = await createEmbedding(openai, submission_text, `course_${course.id}`);
+  // Insert new embedding into the table and return the new embedding
   const new_submission_embedding = await queryRow(
     sql.create_embedding_for_submission,
     {
       embedding: vectorToString(embedding),
       submission_id: submission.id,
-      submission_text: student_answer,
+      submission_text,
       assessment_question_id: instance_question.assessment_question_id,
     },
     SubmissionGradingContextEmbeddingSchema,
   );
   return new_submission_embedding;
 }
 
+function assertRubricNotModified(
+  old_rubric_items: RubricItem[],
+  new_rubric_items: RubricItem[],
+): void {
+  if (old_rubric_items.length !== new_rubric_items.length) {
+    throw new Error('rubric modified between rubric retrieval and score insertion');
+  }
+  for (let i = 0; i < old_rubric_items.length; i++) {
+    const old_item = old_rubric_items[i];
+    const new_item = new_rubric_items[i];
+    if (
+      old_item.description !== new_item.description ||
+      old_item.explanation !== new_item.explanation
+    ) {
+      throw new Error('rubric modified between rubric retrieval and score insertion');
```

Comment:
Rather than throwing an error, could we just do our best and fill in as much of the new rubric as possible? This is essentially a race condition -- if we'd got the grading result back just slightly earlier we wouldn't have had a problem, and then the rubric modification would have occurred anyway. I think the user can't expect anything miraculous no matter which way the race resolves, so we may as well just do the best we can and not error.